types/wlr_output_layout.c

```diff
@@ -103,26 +131,120 @@ void wlr_output_layout_move(struct wlr_output_layout *layout,
 
 void wlr_output_layout_remove(struct wlr_output_layout *layout,
 		struct wlr_output *output) {
-	struct wlr_output_layout_output *layout_output =
-			wlr_output_layout_get(layout, output);
-	if (layout_output) {
-		wl_list_remove(&layout_output->link);
-		free(layout_output);
+	struct wlr_output_layout_output *l_output;
+	l_output= wlr_output_layout_get(layout, output);
+	if (l_output) {
+		wlr_output_layout_output_destroy(l_output);
 	}
 }
 
 void wlr_output_layout_output_coords(struct wlr_output_layout *layout,
-		struct wlr_output *reference, int *x, int *y) {
+		struct wlr_output *reference, double *x, double *y) {
 	assert(layout && reference);
-	int src_x = *x;
-	int src_y = *y;
+	double src_x = *x;
+	double src_y = *y;
 
 	struct wlr_output_layout_output *_output;
 	wl_list_for_each(_output, &layout->outputs, link) {
 		if (_output->output == reference) {
-			*x = src_x - _output->x;
-			*y = src_y - _output->y;
+			*x = src_x - (double)_output->x;
+			*y = src_y - (double)_output->y;
 			return;
 		}
 	}
 }
+
+static double get_distance(double x1, double y1, double x2, double y2) {
+	double distance;
+	distance = sqrt((x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2));
```

Comment:
If we just want the distance to compare, just work with squared-distance as sqrt preserve ordering.
Ignore me if it's used as distance proper.

Reply:
You can use hypot from math.h instead of doing this.